boards/arm/nrf52/nrf52840-dk/src/nrf52_sx127x.c

```diff
@@ -88,7 +88,7 @@ static int sx127x_irq0_attach(xcpt_t isr, FAR void *arg)
 
   /* IRQ on rising edge */
 
-  nrf52_gpiosetevent(GPIO_SX127X_DIO0, true, false, false, isr, arg);
+  nrf52_gpiote_set_pin_event(GPIO_SX127X_DIO0, isr, arg);
```

Comment:
I checked it and it doesn't work. 
`nrf52_gpiote_isr` is only called somewhere on startup where it shouldn't be (probably because of `nrf52_gpio_sense` in `nrf52_gpio_config`)

 Changed to `nrf52_gpiote_set_ch_event(GPIO_SX127X_DIO0, 0, true, false, isr, arg);` and it triggers IRQ.

Reply:
Can you check if this pin is already HIGH on boot? In that case the interrupt will fire immediately as you say. In such case it may indeed be better to use a GPIOTE channel as it directly supports edge sensitive trigger.

Reply:
Yes, this pin is HIGH on boot.

Reply:
Ok, so this could then be done using a GPIOTE channel. I can change it if you prefer.
Note that it can be handled with a per-pin callback with the PORT event but requires more work to make it edge-sensitive, it would require:
- setting SENSE to LOW initially
- setup the pin callback to toggle the SENSE polarity on each call
- only react to relevant edge (in this case, low-to-high, so it would be done when entering callbacking with SENSE set to HIGH).

This toggling is annoying, but is actually considered in nordic SDK (you pass a NRF_GPIOTE_POLARITY_TOGGLE flag, and on each call it toggles the sense polarity) to handle this case. The motivation to do it this way is to not use GPIOTE unless necessary, since it is a scarce resource and it uses more current (if SPI is active, it is about 400uA due to chip errata present in all chips).

Reply:
I changed this to use a channel